src/guacscanner/guacscanner.py

```diff
@@ -278,8 +279,8 @@ def get_entity_id(db_connection, entity_name, entity_type):
 def add_user(
     db_connection: psycopg.Connection,
     username: str,
-    password: str = None,
-    salt: bytes = None,
+    password: typing.Optional[str] = None,
+    salt: typing.Optional[bytes] = None,
```

Comment:
```suggestion
    password: Optional[str] = None,
    salt: Optional[bytes] = None,
```

Reply:
In this case both `typing.Optional` and `schema.Optional` are floating around and I can't have them clobbering each other.

Reply:
I would prefer we switch the `schema` import to `import schema` and prefix the pieces we use and leave the `from typing import Optional`. The `typing` imports are more likely to be interspersed in the code base while the `schema` stuff will be isolated to where we do argument validation.

Reply:
Please see comment 0b35318.

Reply:
Thanks!